src/datastore/datastore.rs

```diff
@@ -762,4 +776,14 @@ impl Datastore {
             Err(e) => Err(e)
         }
     }
+
+    pub fn commit(&self) -> Result<(), DatastoreError> {
```

Comment:
Name this force_commit to make it clear that it is usually not necessary.
You could even add a sentence of why it exists.